### PR TITLE
Fix fixer for jsx-sort-comp in case of more than 10 props

### LINKED ISSUE #1922: autofix breaking all tags
Hello.
I can not say, confidently, that this is a bug or a problem with my code. But when I run eslint with --fix the lines under `jsx-sort-props` is breaking all the file and making such a mess.

Here is the output only to this file without the `--fix` flag:

```
 5:1  error  Imports should be sorted alphabetically  sort-imports-es6-autofix/sort-imports-es6
  118:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  119:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  120:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  121:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  122:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  130:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  132:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  133:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  134:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  135:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  136:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  137:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  138:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  149:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  150:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  151:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  159:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  161:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  162:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  163:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  164:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  165:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  166:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  167:9  error  Props should be sorted alphabetically    react/jsx-sort-props
  178:9  error  Props should be sorted alphabetically    react/jsx-sort-props
```

Ok. When I run the --fix flag, the JSX Tags in my file turn from this:

```jsx
<CreateNewJob
        closed={false}
        flagOptions={flagOptions}
        jobHeight={300}
        jobWidth={200}
        campaign='Some Campaign name'
        campaignStart={moment('2018-07-28 00:00:00')}
        campaignFinish={moment('2018-09-01 00:00:00')}
        jobNumber={'Job Number can be a String'}
        jobTemplateOptions={jobTemplateOptions}
        numberOfPages={30}
        onChange={onChange}
        onClose={onClose}
        spreadSheetTemplateOptions={spreadSheetTemplateOptions}
        stateMachineOptions={stateMachineOptions}
        workflowTemplateOptions={workflowTemplateOptions}
        workflowTemplateSteps={workflowTemplateSteps}
        description='Some description for this job'

        jobTemplate='1'
        stateMachine='1'
        flag='1'
        spreadSheetTemplate='1'
        workflowTemplate='1'
        validation={validation}
        onSubmit={onSubmit}
      />
```

To this:

```jsx
<CreateNewJob
        campaign='Some Campaign name'
        campaignFinish={moment('2018-09-01 00:00:00')}
        campaignStart={moment('2closed={false}0:00')}
 description='Some description for thisflag='1'ampaignStart={moment('2018-07-28 00:00:00')}
 flagOptions={flagOptions}0:00')}
        jobNumber={'JojobHeight={300}mplateOptions={jobTemplateOptions}jobNumber={'Job Number can be a String'}hange={ojobTemplate='1'nClose={onCljobTemplateOptions={jobTempljobWidth={200}s={30}
        stateMachineOpti
        stateMachineOptions={stateMachineOptions}
        workflowTemplateOptions={workflowTemplateOptions}
        workflowTemplateSteps={workflowTemplateSteps}
        description='Some description for this job'

        jobTemplate='1'
        stateMachine='1'
        flag='1'
        spreadSheetTemplate='1'
        workflowTemplate='1'
        validation={validation}
        onSubmit={onSubmit}
      />
```
Is there anything I could do to make it auto fix work? It would be really tedious to fix them all manually.

Any suggestion is appreciated.

Thank you.

### LINKED ISSUE #1993: react/jsx-sort-props breaks the code
**Tell us about your environment**

    "babel-eslint": "^9.0.0",
    "eslint": "5.6.0",
    "eslint-plugin-react": "^7.11.1"

* **ESLint Version:** 9.0.0
* **Node Version:** 10.6.0
* **npm Version:** -
* **yarn Version:** 1.9.4

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
I created a repo to reproduce the bug: https://github.com/davesnx/eslint-bug-react-sort-props

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  plugins: ['react'],
  rules: {
    'react/jsx-sort-props': 1
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      className={className}
      onStageAnswer={onStageAnswer}
      onCommitAnswer={onCommitAnswer}
      isFocused={isFocused}
      direction={direction}
      allowMultipleSelection={allowMultipleSelection}
      measureLongestChildNode={measureLongestChildNode}
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice

```

```bash
npx eslint . --fix
```

**What did you expect to happen?**
So theorically the rule should re-order my props, in the worst case scenario, not sort them. But actually is breaking them. It's a weird output:

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      allowMultipleSelection={allowMultipleSelection}
      className={className}
      direction={direction}
      isFocused=handleAppScroll={handleAppScroll}ection=isActive={isActive}owMultiplisFocused={isFocused}ardShortcutType
      layoutItemsSize={layoutItemsSl
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice
```

It's actually trying to sort in a weird way, and it's not consistent.
Happens with a boolean props as well

```js

      isFocused=
      
      handleAppScroll={handleAppScroll}
      
      ection=
      
      isActive={isActive}
      
      owMultiplisFocused={isFocused}
      
      ardShortcutType
      layoutItemsSize={layoutItemsSl
      
```

### PR DESCRIPTION
**Pull request by tihonove**
Just a little bug. Fixes #1922. Fixes #1993

### CODE REVIEW

File: lib/rules/jsx-sort-props.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ const generateFixerFunction = (node, context, reservedList) => {
       });
     });
 
-    fixers.sort((a, b) => a.range[0] < b.range[0]);
+    fixers.sort((a, b) => Math.sign(b.range[0] - a.range[0]));
```
Comment (ljharb) [inline lib/rules/jsx-sort-props.js]: I’m not sure the Math.sign is needed at all; comparators can return any negative or positive number.

**Comment by tihonove** [comment]
> Could you add the test cases from both linked issues?
Currently tests contains reduced cases. Do you propose create test cases with exatly same code from issues?

**Comment by ljharb** [comment]
Yes, auto fixing is pretty subtle, and it allows for complete confidence that it avoids the regression. “Too many tests” is a good problem to have :-)

**Comment by tihonove** [comment]
Fix commit message

File: lib/rules/jsx-sort-props.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ const generateFixerFunction = (node, context, reservedList) => {
       });
     });
 
-    fixers.sort((a, b) => a.range[0] < b.range[0]);
+    fixers.sort((a, b) => b.range[0] - a.range[0]);
```
Comment (alexzherdev) [inline lib/rules/jsx-sort-props.js]: Woah great fix here. @tihonove could you explain how this fix works?

Comment (ljharb) [reply lib/rules/jsx-sort-props.js]: Comparators are supposed to only return a negative, positive, or zero number - previously it returned a boolean here.

**Pull request merged**